src/services/sorting.ts

```diff
@@ -1,35 +1,50 @@
 import { ChangepointDetector } from '@bsull/augurs';
 import { DataFrame, FieldType, doStandardCalcs, fieldReducers } from '@grafana/data';
+import { getLabelValueFromDataFrame } from './levels';
+import { memoize } from 'lodash';
 
-export const sortSeries = (series: DataFrame[], sortBy: string, direction: string) => {
-  const reducer = (dataFrame: DataFrame) => {
-    if (sortBy === 'changepoint') {
-      return calculateDataFrameChangepoints(dataFrame);
-    }
-    const fieldReducer = fieldReducers.get(sortBy);
-    const value =
-      fieldReducer.reduce?.(dataFrame.fields[1], true, true) ?? doStandardCalcs(dataFrame.fields[1], true, true);
-    return value[sortBy] ?? 0;
-  };
-
-  const seriesCalcs = series.map((dataFrame) => ({
-    value: reducer(dataFrame),
-    dataFrame: dataFrame,
-  }));
-
-  seriesCalcs.sort((a, b) => {
-    if (a.value !== undefined && b.value !== undefined) {
-      return b.value - a.value;
+export const sortSeries = memoize(
+  (series: DataFrame[], sortBy: string, direction: string) => {
+    const reducer = (dataFrame: DataFrame) => {
+      if (sortBy === 'changepoint') {
+        return calculateDataFrameChangepoints(dataFrame);
+      }
+      const fieldReducer = fieldReducers.get(sortBy);
+      const value =
+        fieldReducer.reduce?.(dataFrame.fields[1], true, true) ?? doStandardCalcs(dataFrame.fields[1], true, true);
+      return value[sortBy] ?? 0;
+    };
+
+    const seriesCalcs = series.map((dataFrame) => ({
+      value: reducer(dataFrame),
+      dataFrame: dataFrame,
+    }));
+
+    seriesCalcs.sort((a, b) => {
+      if (a.value !== undefined && b.value !== undefined) {
+        return b.value - a.value;
+      }
+      return 0;
+    });
+
+    if (direction === 'asc') {
+      seriesCalcs.reverse();
     }
-    return 0;
-  });
 
-  if (direction === 'asc') {
-    seriesCalcs.reverse();
+    return seriesCalcs.map(({ dataFrame }) => dataFrame);
+  },
+  (series: DataFrame[], sortBy: string, direction: string) => {
```

Comment:
Might be a bit too much, but I want to be sure the sorting function is actually being called when a different time series is passed.